src/test/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ProcedureRequestMapperTest.java

```diff
@@ -1,20 +1,283 @@
+        final Message message = new Message(List.of(
+            "S02+02", // ServiceReportDetails
+            "S06+06", // InvestigationSubject
+            "S10+10",
+            "CIN+UN",
+            "FTX+CID+++COELIAC"
+        ));
+
+        final Optional<ProcedureRequest> procedureRequest = procedureRequestMapper.mapToProcedureRequest(
+            message, null, null, null, null, null);
+
+        assertThat(procedureRequest).isNotEmpty();
+        assertThat(procedureRequest.get().getStatus().toCode()).isEqualTo("unknown");
+    }
+
+    @Test
+    void testMapToProcedureRequestUndefinedStatus() {
+        final Message message = new Message(List.of(
+            "S02+02", // ServiceReportDetails
+            "S06+06", // InvestigationSubject
+            "S10+10",
+            "CIN+Undefined Status",
+            "FTX+CID+++COELIAC"
+        ));
+
+        assertThatThrownBy(() -> procedureRequestMapper.mapToProcedureRequest(
+            message, null, null, null, null, null))
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessage("No Report Status Code for 'Undefined Status'");
+    }
+
+    @Test
+    void testMapToProcedureRequestIntent() {
+        final Message message = new Message(List.of(
+            "S02+02", // ServiceReportDetails
+            "S06+06", // InvestigationSubject
+            "S10+10",
+            "CIN+UN",
+            "FTX+CID+++COELIAC"
+        ));
+
+        final Optional<ProcedureRequest> procedureRequest = procedureRequestMapper.mapToProcedureRequest(
+            message, null, null, null, null, null);
+
+        assertThat(procedureRequest).isNotEmpty();
+        assertThat(procedureRequest.get().getIntent()).isEqualTo(ProcedureRequestIntent.NULL);
+    }
+
+    @Test
+    void testMapToProcedureRequestPatientReference() {
+        final Message message = new Message(List.of(
+            "S02+02", // ServiceReportDetails
+            "S06+06", // InvestigationSubject
+            "S10+10",
+            "CIN+UN",
+            "FTX+CID+++COELIAC"
+        ));
+        when(fullUrlGenerator.generate(any(Patient.class))).thenReturn("patient-full-url");
+
+        final Optional<ProcedureRequest> procedureRequest = procedureRequestMapper.mapToProcedureRequest(
+            message, mock(Patient.class), null, null, null, null);
+
+        assertThat(procedureRequest)
+            .isNotEmpty()
+            .hasValueSatisfying(procedure ->
+                assertThat(procedure.getSubject().getReference()).isEqualTo("patient-full-url"));
+    }
+
+    @Test
+    void testMapToProcedureRequestRequesterReferenceWhenPerformingOrganisationIsPresented() {
+        final Message message = new Message(List.of(
+            "S02+02", // ServiceReportDetails
+            "S06+06", // InvestigationSubject
+            "S10+10",
+            "CIN+UN",
+            "FTX+CID+++COELIAC"
+        ));
+        lenient().when(fullUrlGenerator.generate(any(Organization.class)))
```

Comment:
Does this really need to be lenient?

Reply:
Otherwise, I get the `Strict stubbing argument mismatch` error. Do you think there are some other ways to deal with that?

Reply:
Looks like it's because the `Organization` is `null` when running the test, and `any()` doesn't match `null`. You can use `nullable(Organization.class)` instead.

Reply:
Oh, that's true! I've removed all other calls to lenient() in this class f30824aa0fa1105faf6da08e191e1ab08419aa82
Thanks!